### ISSUE TITLE
route match problem

### ISSUE THREAD

**Issue by ccbikai** [fastify/fastify#573]
```
const fastify = require('fastify')

const app = fastify()

app.get('/:slug/', (req, res) => {
  res.send('right')
})

app.get('/amp/:slug/', (req, res) => {
  res.send('amp')
})

app.get('*', (req, res) => {
  res.send('wrong')
})

app.listen(3333)
```

![image](https://user-images.githubusercontent.com/2959393/34293136-84047f66-e73e-11e7-9520-adcf38fd9d6b.png)

when i `curl "http://localhost:3333/a/"`, it got 'wrong'.

I think should be 'right'

**Comment by delvedor** [comment]
Hi!
If I'm not wrong, `/` means a new level of url nesting, so declare a route as `/test/` does not have sense, and the wildcard will correctly catch all the not matching routes.
@mcollina thoughts?

**Comment by ccbikai** [comment]
but i `curl "http://localhost:3333/b/"`, it got 'right'


![image](https://user-images.githubusercontent.com/2959393/34305242-176e5ff4-e778-11e7-8ae5-711b901a0738.png)

**Comment by ccbikai** [comment]
When i use express, it's all 'right'

```
const fastify = require('express')

const app = fastify()

app.get('/:slug/', (req, res) => {
  res.send('right')
})

app.get('/amp/:slug/', (req, res) => {
  res.send('amp')
})

app.get('*', (req, res) => {
  res.send('wrong')
})

app.listen(3333)
```
![image](https://user-images.githubusercontent.com/2959393/34305353-b75b0c24-e778-11e7-9c87-0a634e2fd51f.png)

**Comment by delvedor** [comment]
Oks, thank you for reporting, I'll investigate on this :)

**Comment by delvedor** [comment]
Fixed in `find-my-way` [`v1.10.1`](https://github.com/delvedor/find-my-way/releases/tag/v1.10.1).
